### ISSUE TITLE
Traceback creating WatchEvent

### ISSUE THREAD

**Issue by jayme-github** [kelproject/pykube#40]
It seems as if the response objects do not have a `type` :

```python
wq = Node.objects(api).all().watch(now)
for e in wq.object_stream():
    print(e)
```

```pytb
site-packages/pykube/query.py in object_stream(self)
    134             we = json.loads(line.decode("utf-8"))
--> 135             yield WatchEvent(type=we["type"], object=self.api_obj_class(self.api, we["object"]))
    136 
    137     def __iter__(self):

KeyError: 'type'
```

//EDIT:
My bad. Looks as `watch()` (at least for Nodes) must be created from a "namespace=all_" filter for the API-URL to be computed correctly:
```py
wq = Node.objects(api).filter(namespace=all_).watch(now)
for e in wq.object_stream():
    print(e)
```

Maybe I just don't know how to use that correctly. :smile:

**Comment by brosner** [comment]
I wonder if this is because it is using the `default` namespace. There are some API objects that are not namespaced and shouldn't ever include a namespace in their API URLs. We may need to fix that as it is slightly annoying to add in `namespace=all_` for these (however, is a current workaround).

**Comment by diegodelemos** [comment]
It is not working when I specify a certain namespace. I just add the call with raw requests to show that this is in fact working and therefore is something related with pykube. Maybe I am using it wrong, but this is something that definitely needs to be supported IMHO
```python
>>> wq = pykube.Job.objects(api).filter(namespace='alice').watch()   
>>> for line in wq:                                               
...     print(line)                                               
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/site-packages/pykube/query.py", line 156, in object_stream
    yield WatchEvent(type=we["type"], object=self.api_obj_class(self.api, we["object"]))
KeyError: 'type'
<('https://kubernetes/apis/extensions/v1beta1/namespaces/alice/jobs', params={>
>>> response = requests.get('https://kubernetes/apis/extensions/v1beta1/namespaces/default/jobs', params={'watch': 'true'}, headers={'Authorization': 'Bearer {}'.format(TOKEN)}, verify=CA_CERT_PATH, stream=True)
>>> for line in response.iter_lines():                                   
...     print(line)
... 
b'{"type":"ADDED","object":{"kind":"Job","apiVersion":"extensions/v1beta1","metadata":{"name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0","namespace":"alice","selfLink":"/apis/extensions/v1beta1/namespaces/alice/jobs/02503793-d1ec-40ad-9a41-cdd22bcdd964-0","uid":"db7cffed-96cd-11e6-b606-5e188a8d8555","resourceVersion":"129309","creationTimestamp":"2016-10-20T14:02:37Z"},"spec":{"parallelism":1,"completions":1,"selector":{"matchLabels":{"controller-uid":"db7cffed-96cd-11e6-b606-5e188a8d8555"}},"autoSelector":true,"template":{"metadata":{"name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0","creationTimestamp":null,"labels":{"controller-uid":"db7cffed-96cd-11e6-b606-5e188a8d8555","job-name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0"}},"spec":{"volumes":[{"name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0","hostPath":{"path":"/data/alice/alice-worker-ncpyv/02503793-d1ec-40ad-9a41-cdd22bcdd964/0"}}],"containers":[{"name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0","image":"docker-img","env":[{"name":"RANDOM_ERROR","value":"1"}],"resources":{},"volumeMounts":[{"name":"02503793-d1ec-40ad-9a41-cdd22bcdd964-0","mountPath":"/data"}],"terminationMessagePath":"/dev/termination-log","imagePullPolicy":"Always"}],"restartPolicy":"Never","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","securityContext":{}}}},"status":{}}}'
```

**Comment by jayme-github** [comment]
This seems fixed to me with #71 and the following patch (but I've not tested extensively):

```patch
diff --git a/pykube/query.py b/pykube/query.py
index 9db1696..ad95b7e 100644
--- a/pykube/query.py
+++ b/pykube/query.py
@@ -139,9 +139,7 @@ class WatchQuery(BaseQuery):
         params = {}
         if self.resource_version is not None:
             params["resourceVersion"] = self.resource_version
-        url = self._build_api_url(params=params)
-        if self.namespace is all_:
-            url = "watch/" + url
+        url = "watch/" + self._build_api_url(params=params)
         kwargs = {
             "url": url,
             "stream": True,
```
